android/src/main/java/sqip/flutter/internal/CardEntryModule.java

```diff
@@ -55,32 +65,63 @@ public CardEntryModule(PluginRegistry.Registrar registrar, final MethodChannel c
-        CardEntry.handleActivityResult(data, new Callback<CardEntryActivityResult>() {
-          @Override public void onResult(final CardEntryActivityResult cardEntryActivityResult) {
-            // flutter UI doesn't know the context of fade_out animation
-            // so that the next action from flutter can be triggered too soon before
-            // card entry activity is closed completely.
-            // So this is a workaround to delay the callback until animation finished.
-            long delayDurationMs = readCardEntryCloseExitAnimationDurationMs();
-            handler.postDelayed(new Runnable() {
-              @Override
-              public void run() {
-                if (cardEntryActivityResult.isCanceled()) {
-                  channel.invokeMethod("cardEntryCancel", null);
-                } else if (cardEntryActivityResult.isSuccess()) {
-                  channel.invokeMethod("cardEntryComplete", null);
-                }
+        if (requestCode == CardEntry.DEFAULT_CARD_ENTRY_REQUEST_CODE) {
+          CardEntry.handleActivityResult(data, new Callback<CardEntryActivityResult>() {
+            @Override
+            public void onResult(final CardEntryActivityResult cardEntryActivityResult) {
+              if (cardEntryActivityResult.isSuccess() && CardEntryModule.this.contact != null) {
+                cardResult = cardEntryActivityResult.getSuccessValue();
+                String paymentSourceId = cardResult.getNonce();
+                VerificationParameters verificationParameters = new VerificationParameters(paymentSourceId, CardEntryModule.this.buyerAction, CardEntryModule.this.squareIdentifier, CardEntryModule.this.contact);
+                BuyerVerification.verify(currentActivity, verificationParameters);
+              } else {
+                // flutter UI doesn't know the context of fade_out animation
+                // so that the next action from flutter can be triggered too soon before
+                // card entry activity is closed completely.
+                // So this is a workaround to delay the callback until animation finished.
+                long delayDurationMs = readCardEntryCloseExitAnimationDurationMs();
+                handler.postDelayed(new Runnable() {
+                  @Override
+                  public void run() {
+                    if (cardEntryActivityResult.isCanceled()) {
+                      channel.invokeMethod("cardEntryCancel", null);
+                    } else if (cardEntryActivityResult.isSuccess()) {
+                      channel.invokeMethod("cardEntryDidObtainCardDetails", cardDetailsConverter.toMapObject(cardEntryActivityResult.getSuccessValue()));
+                    }
+                  }
+                }, delayDurationMs);
               }
-            }, delayDurationMs);
-          }
-        });
+            }
+          });
+        }
+
+        if (requestCode == BuyerVerification.DEFAULT_BUYER_VERIFICATION_REQUEST_CODE) {
```

Comment:
should use `else if`?

Reply:
Hmm I'm ok either way. Here the style using two `if` as opposed to `else if` is made to be consistent with step 3.4 in https://developer.squareup.com/docs/in-app-payments-sdk/verify-buyer.